packages/core/test/foundry/optimistic-assertor/WhitelistedSovereignSecurityManager.t.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "forge-std/Test.sol";
+import "../fixtures/common/TestAddress.sol";
+import "../../../contracts/optimistic-assertor/implementation/sovereign-security-manager/WhitelistedSovereignSecurityManager.sol";
+import "../../../contracts/optimistic-assertor/interfaces/OptimisticAssertorInterface.sol";
+
+contract WhitelistedSovereignSecurityManagerTest is Test {
+    WhitelistedSovereignSecurityManager ssm;
+    address mockOptimisticAssertorAddress = address(0xff);
+    address mockAssertingCallerAddress = address(0xffa);
+
+    function setUp() public {
+        ssm = new WhitelistedSovereignSecurityManager();
+    }
+
+    function test_AssertingCallerWhitelist() public {
+        bytes32 assertionId = "test";
+
+        // If the asserting caller is not whitelisted, then the assertion should not be allowed.
+        _mockReadAssertionAssertingCaller(mockAssertingCallerAddress);
+        vm.prank(mockOptimisticAssertorAddress);
+        SovereignSecurityManagerInterface.AssertionPolicies memory policyNotWhitelisted =
+            ssm.getAssertionPolicies(assertionId);
+        assertEq(policyNotWhitelisted.allowAssertion, false);
+
+        // If the asserting caller is whitelisted, then the assertion should be allowed.
+        ssm.setAssertingCallerInWhitelist(mockAssertingCallerAddress, true);
+        _mockReadAssertionAssertingCaller(mockAssertingCallerAddress);
```

Comment:
No need to repeat the same mock as "Mocked calls are in effect until `clearMockedCalls` is called"